scipy/special/cephes/incbet.c

```diff
@@ -72,15 +72,17 @@ static double incbcf(double a, double b, double x);
 static double incbd(double a, double b, double x);
 static double pseries(double a, double b, double x);
 
-double incbet(aa, bb, xx)
-double aa, bb, xx;
+double incbet(double aa, double bb, double xx)
 {
     double a, b, t, x, xc, w, y;
     int flag;
 
-    if (aa <= 0.0 || bb <= 0.0)
+    if (aa < 0.0 || bb < 0.0)
 	goto domerr;
 
+    if (aa == 0.0 || bb == 0.0)
```

Comment:
This can't be quite right-now you're going to get

```python
sc.betainc(1e-300, 1e-300, 0) == 0
```

but

```python
sc.betainc(0, 0, 0) == 1
```

i.e. there's a discontinuity, which should result in NaN.

I think we're also going to need some justification that this does the right thing for `a == 0` but `b == inf` say. Generally speaking there's a cube

```
a         b         x
[0, oo] x [0, oo] x [0, 1]
```

and each of the 6 faces, 12 edges, and 8 vertices needs to be thought about.

Reply:
@person142  Thanks! Your right, I'd been thinking this was just a fix for an isolated point but will require some more thinking, and maybe I need to look at a contour integral or two to understand the behavior along the faces of this space.

Reply:
There might be a consistent way to define the values for `a == 0, b > 0` or for `a > 0, b == 0`, but for `a == 0, b == 0`, the value is not well-defined.  You can see this numerically by computing the function for a series of parameter values (a(t), b(t)) that approach (0, 0) along different straight lines:

```
In [140]: t                                                                               
Out[140]: array([1.e-06, 1.e-07, 1.e-08, 1.e-09, 1.e-10, 1.e-11, 1.e-12, 1.e-13])

In [141]: betainc(t, t, 0.3)
Out[141]: 
array([0.49999958, 0.49999996, 0.5       , 0.5       , 0.5       ,
       0.5       , 0.5       , 0.5       ])

In [142]: betainc(t, 3*t, 0.3)
Out[142]: 
array([0.74999936, 0.74999994, 0.74999999, 0.75      , 0.75      ,
       0.75      , 0.75      , 0.75      ])

In [143]: betainc(2*t, 3*t, 0.3)
Out[143]: 
array([0.59999898, 0.5999999 , 0.59999999, 0.6       , 0.6       ,
       0.6       , 0.6       , 0.6       ])
```

Reply:
I think we're all in agreement on this one, consistent values here:
`a == 0, b > 0, x > 0` -> 1
`a > 0, b == 0, x > 0` ->  1
`a == 0, b == 0, x>= 0` -> `Nan`

I know Josh would like the `inf` cases handled too but not clear to me the limit value(s) for those cases so calling those  'out of scope' for this PR.